setuptools/build_meta.py

```diff
@@ -158,26 +94,103 @@ def _file_with_extension(directory, extension):
     return file
 
 
-def build_wheel(wheel_directory, config_settings=None,
-                metadata_directory=None):
-    config_settings = _fix_config(config_settings)
-    wheel_directory = os.path.abspath(wheel_directory)
-    sys.argv = sys.argv[:1] + ['bdist_wheel'] + \
-        config_settings["--global-option"]
-    _run_setup()
-    if wheel_directory != 'dist':
-        shutil.rmtree(wheel_directory)
-        shutil.copytree('dist', wheel_directory)
+class _BuildMetaBackend(object):
 
-    return _file_with_extension(wheel_directory, '.whl')
+    def _fix_config(self, config_settings):
+        config_settings = config_settings or {}
+        config_settings.setdefault('--global-option', [])
+        return config_settings
 
+    def _get_build_requires(self, config_settings, requirements):
+        config_settings = self._fix_config(config_settings)
 
-def build_sdist(sdist_directory, config_settings=None):
-    config_settings = _fix_config(config_settings)
-    sdist_directory = os.path.abspath(sdist_directory)
-    sys.argv = sys.argv[:1] + ['sdist', '--formats', 'gztar'] + \
-        config_settings["--global-option"] + \
-        ["--dist-dir", sdist_directory]
-    _run_setup()
-
-    return _file_with_extension(sdist_directory, '.tar.gz')
+        sys.argv = sys.argv[:1] + ['egg_info'] + \
+            config_settings["--global-option"]
```

Comment:
a line apart using both ``'`` and ``"`` to represent strings?

Reply:
I think this is not actually new code - I wouldn't use a line continuation either.

At some point we'll run `black` on the whole repo, when we get our act together about enforcement.